subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -142,6 +142,42 @@ Only secure protocols like `https` are accepted.
 This is required to make sure no one can tamper with the download in flight.
 ====
 
+[[sub:viewing_toolchains]]
+=== Viewing and debugging toolchains
+
+Gradle can visualize the list of all detected toolchains including their metadata.
+
+For example, to show all toolchains of a project, run:
+
+```
+gradle -q javaToolchains
+```
+
+.Output of **`gradle -q javaToolchains`**
+----
+> gradle -q javaToolchains
+
+ + AdoptOpenJDK 1.8.0_242
+     | Location:           /Users/username/.sdkman/candidates/java/8.0.242.hs-adpt/jre
+     | Language Version:   8
+     | Is JDK:             true
+     | Detected by:        SDKMAN!
+
+ + OpenJDK 15-ea
+     | Location:           /Users/username/.sdkman/candidates/java/15.ea.21-open
+     | Language Version:   15
+     | Is JDK:             true
+     | Detected by:        SDKMAN!
+
+ + Oracle JDK 1.7.0_80
+     | Location:           /Library/Java/JavaVirtualMachines/jdk1.7.0_80.jdk/Contents/Home/jre
+     | Language Version:   7
+     | Is JDK:             true
+     | Detected by:        macOS java_home
+----
+
+This can help to debug which toolchains are available to the build, how they are detected and what kind of metadata Gradle knows about those toolchains.
```

Comment:
Maybe you could describe what is displayed for toolchains that are passed through env vars or references to their java home directly?

Reply:
Good idea! Done